### PR TITLE
PLA-232 Switch nextjs template to app router

### PR DESCRIPTION
**Pull request by quesabe**
Closes PLA-232.

The following changes done:
- moved pages to _app_ folder (and changed lint paths accordingly);
- extended _tsconfig_ (added `next` plugin into, added next types to _include_ array);
- added app folder to _tailwindcss_ config.

### CODE REVIEW

**Review by gvidon — COMMENTED** [review]
@quesabe are these breaking changes? Do we need to bump the minor version of the release?

**Comment by quesabe** [comment]
> @quesabe are these breaking changes? Do we need to bump the minor version of the release?

I don't think so. The main reason for this is the fact that nextjs recommends [gradual adoption of app router](https://nextjs.org/docs/app/building-your-application/upgrading/app-router-migration). I've tried it on the sample files - it seems working.

However I can't guarantee it with 100% confidence 😳

BTW I planned a communication, that we switch the template to app router and some guides on how to opt out from it. I did some testing - it seems like one can just disable sample code generation with an option and stay on pages router (the changes to _tsconfig_ and _tailwind_ config do not break compatibility with routing based on `pages` folder).

Do you suggest asking project owners to test the changes manually before committing this update to the template?

**Comment by gvidon** [comment]
@quesabe yeah, let's do a canary release and ask @AleksandrIvanenko and @Serator to test.

**Comment by gvidon** [comment]
@AleksandrIvanenko did you have a chance to test canary release?

**Comment by AleksandrIvanenko** [comment]
> @AleksandrIvanenko did you have a chance to test canary release?

@gvidon, sure, it's already applied in Metrus. We discussed it [here](https://ottofeller.slack.com/archives/C041U3KV1T7/p1687115912446399).

File: src/nextjs/index.ts [hunk 1]
```diff
@@ -108,7 +113,7 @@ export class OttofellerNextjsProject extends NextJsTypeScriptProject {
     new SampleFile(this, 'next-env.d.ts', {sourcePath: path.join(assetsDir, 'next-env.d.ts.sample')})
 
     // ANCHOR ESLint and prettier setup
-    const lintPaths = options.lintPaths ?? ['.projenrc.ts', 'pages', 'src']
+    const lintPaths = options.lintPaths ?? ['.projenrc.ts', 'app', 'src']
```
Comment (AleksandrIvanenko) [inline src/nextjs/index.ts:116]: @quesabe, I think we can leave `pages` folder in case there is a incremental adoption applied somewhere

Comment (quesabe) [reply src/nextjs/index.ts:116]: I though that any intermediate state  (e.g. during incremental migration) is better be left to a user - the `lintPaths` helps configuring it manually. This way we can leave new projects with a cleaner structure. In addition there is a possibility for `lint` command to fail if `pages` folder does not exist.

Overall I targeted mostly users who spin up new projects. But there are ways to handle both pages router, app router, and something in between - and I planned to add a migration guide to the release notes.

**Pull request merged**